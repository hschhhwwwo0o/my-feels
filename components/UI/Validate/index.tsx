import React, { FunctionComponent } from "react";

const Validate: FunctionComponent<{ isValidate: boolean }> = ({ children, isValidate = true }) => {
  return (
    <div className={`${isValidate ? "opacity-100" : "opacity-60 pointer-events-none"} transition-opacity duration-500`}>
      {children}
    </div>
  );
};

export default Validate;
