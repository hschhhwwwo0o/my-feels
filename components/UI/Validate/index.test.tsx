import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Validate from "./index";

test("Test Validate component with true", () => {
  render(
    <Validate data-testid="validate" isValidate={true}>
      Ok
    </Validate>
  );
  const el: HTMLElement = screen.getByText("Ok");
  expect(el).toBeInTheDocument();
  expect(el).toHaveClass("opacity-100");
});

test("Test Validate component with false", () => {
  render(
    <Validate data-testid="validate" isValidate={false}>
      Ok
    </Validate>
  );
  const el: HTMLElement = screen.getByText("Ok");
  expect(el).toBeInTheDocument();
  expect(el).toHaveClass("opacity-60");
});

test("Snapshot Validate component with false", () => {
  render(
    <Validate data-testid="validate" isValidate={false}>
      Ok
    </Validate>
  );
  const el: HTMLElement = screen.getByText("Ok");
  expect(el).toBeInTheDocument();
  expect(el).toMatchSnapshot();
});
