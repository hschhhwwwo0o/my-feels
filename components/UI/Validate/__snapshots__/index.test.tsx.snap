// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Validate component with false 1`] = `
<div
  class="opacity-60 pointer-events-none transition-opacity duration-500"
>
  Ok
</div>
`;
